sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/ReduceNumShufflePartitions.scala

```diff
@@ -91,14 +92,13 @@ case class ReduceNumShufflePartitions(conf: SQLConf) extends Rule[SparkPlan] {
       val distinctNumPreShufflePartitions =
         validMetrics.map(stats => stats.bytesByPartitionId.length).distinct
       if (validMetrics.nonEmpty && distinctNumPreShufflePartitions.length == 1) {
-        val partitionStartIndices = estimatePartitionStartIndices(validMetrics.toArray)
-        // This transformation adds new nodes, so we must use `transformUp` here.
+        val excludedPartitions =
+          ShuffleQueryStageExec.getShuffleStage(shuffleStages.head).excludedPartitions
```

Comment:
`shuffleStages` is `Seq[ShuffleQueryStageExec]`, we can just do `shuffleStages.head.excludedPartitions`